projects/lib/src/lib/index.ts

```diff
@@ -153,14 +158,21 @@ function createStateSlice(
   );
 }
 
-export const syncStateUpdate = (
-  state: any,
-  keys: Keys,
-  storage: Storage,
-  storageKeySerializer: (key: string | number) => string,
-  removeOnUndefined: boolean,
-  syncCondition?: (state: any) => any,
-) => {
+export const syncStateUpdate = ({
+  state,
+  keys,
+  storage,
+  storageKeySerializer,
+  removeOnUndefined,
+  syncCondition,
+}: {
+  state: any;
+  keys: Keys;
+  storage: Storage;
+  storageKeySerializer: (key: string) => string;
```

Comment:
Changed the `key` prop to only be a `string` rather than `string | number`. The public API remains unchanged.